packages/fcl-ethereum-provider/src/accounts/account-manager.ts

```diff
@@ -98,4 +107,86 @@ export class AccountManager {
       callback(this.getAccounts())
     })
   }
+
+  async sendTransaction({
+    to,
+    value,
+    data,
+    gasLimit,
+    chainId,
+  }: {
+    to: string
+    value: string
+    data: string
+    nonce: string
+    gasLimit: string
+    chainId: string
+  }) {
+    // Find the Flow network based on the chain ID
+    const flowNetwork = Object.entries(FLOW_CHAINS).find(
+      ([, chain]) => chain.eip155ChainId === parseInt(chainId)
+    )?.[0] as FlowNetwork
+
+    const evmContractAddress = fcl.withPrefix(
+      FLOW_CONTRACTS[ContractType.EVM][flowNetwork]
+    )
+
+    const txId = await fcl.mutate({
+      cadence: `import EVM from ${evmContractAddress}
+        
+        /// Executes the calldata from the signer's COA
+        ///
+        transaction(evmContractAddressHex: String, calldata: String, gasLimit: UInt64, value: UInt256) {
+        
+            let evmAddress: EVM.EVMAddress
+            let coa: auth(EVM.Call) &EVM.CadenceOwnedAccount
+        
+            prepare(signer: auth(BorrowValue) &Account) {
+                self.evmAddress = EVM.addressFromString(evmContractAddressHex)
+        
+                self.coa = signer.storage.borrow<auth(EVM.Call) &EVM.CadenceOwnedAccount>(from: /storage/evm)
+                    ?? panic("Could not borrow COA from provided gateway address")
+            }
+        
+            execute {
+                let valueBalance = EVM.Balance(attoflow: value)
+                let callResult = self.coa.call(
+                    to: self.evmAddress,
+                    data: calldata.decodeHex(),
+                    gasLimit: gasLimit,
+                    value: valueBalance
+                )
+                assert(callResult.status == EVM.Status.successful, message: "Call failed")
+            }
+        }`,
+      limit: 9999,
+      args: (arg: typeof fcl.arg, t: typeof fcl.t) => [
+        arg(to, t.String),
+        arg(data, t.String),
+        arg(gasLimit, t.UInt64),
+        arg(value, t.UInt256),
+      ],
+      authz: this.user,
+    })
+
+    const result = await fcl.tx(txId).onceExecuted()
+    const {events} = result
+
+    const evmTxExecutedEvent = events.find(
+      event =>
+        event.type ===
+        EVENT_IDENTIFIERS[EventType.TRANSACTION_EXECUTED][flowNetwork]
+    )
+    if (!evmTxExecutedEvent) {
+      throw new Error("EVM transaction hash not found")
+    }
+
+    const eventData: TransactionExecutedEvent = evmTxExecutedEvent.data
+    const evmTxHash = eventData.hash.reduce(
```

Comment:
`const evmTxHash = eventData.hash.map(val => val.padStart(2, "0")).join("")` is more efficient.